habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py

```diff
@@ -32,6 +32,23 @@
 from typing import List, Tuple
 
 
+def bin_search(v, a, b, target_value):
+    left = a
+    right = b
+
+    while left <= right:
+        mid = (left + right) // 2
+
+        if v[mid] == target_value:
+            return mid
+        elif v[mid] < target_value:
+            left = mid + 1
+        else:
+            right = mid - 1
+
+    return min(left, len(v) - 1)
```

Comment:
It would be worthwhile unit testing this one.

We could also use [bisect](https://docs.python.org/3.11/library/bisect.html) to solve this.